extensions/amp-image-slider/0.0/amp-image-slider.js

```diff
@@ -0,0 +1,566 @@
+    const leftPercentage = (e.pageX - left) / width;
+    this.animateUpdatePositions(leftPercentage);
+  }
+
+  /**
+   * Handle tap on the image
+   * @param {Event} e
+   */
+  handleTapImage(e) {
+    // const {left, width} = this.getLayoutBox();
+    const {left, width} = this.container_./*OK*/getBoundingClientRect();
+    if (e.touches.length > 0) {
+      const leftPercentage = (e.touches[0].pageX - left) / width;
+      this.animateUpdatePositions(leftPercentage);
+    }
+  }
+
+  /**
+   * Handle hinding the hint
+   * @param {Event} unusedEvent
+   */
+  handleHideHint(unusedEvent) {
+    if (!this.isBarHintHidden_ && this.barHint_) {
+      this.isBarHintHidden_ = true;
+      this.barHint_.classList.add('i-amphtml-image-slider-hint-hidden');
+    }
+  }
+
+  /**
+   * Animated, update element positions based on percentage
+   * @param {number} toPercentage
+   */
+  animateUpdatePositions(toPercentage) {
+    const {left: containerLeft, width: containerWidth}
+    //    = this.getLayoutBox();
+        = this.container_./*OK*/getBoundingClientRect();
+    const {left: barLeft} = this.bar_./*OK*/getBoundingClientRect();
+
+    const fromPercentage = (barLeft - containerLeft) / containerWidth;
+    const interpolate = numeric(fromPercentage, toPercentage);
+    const curve = bezierCurve(0.4, 0, 0.2, 1); // fast-out-slow-in
+    const duration = 200;
+    // Single animation ensure elements have props updated at same pace
+    // Multiple animations would be scheduled at different raf
+    return Animation.animate(this.element, pos => {
+      this.updatePositions(interpolate(pos));
+    }, duration, curve).thenAlways();
+  }
+
+  /**
+   * Add listeners on drag start
+   * @param {Event} e
+   */
+  dragStart(e) {
+    e.preventDefault();
+    e.stopPropagation();
+
+    this.win.addEventListener('mousemove', this.dragMove);
+    this.win.addEventListener('mouseup', this.dragEnd);
+
+    this.moveOffset_ = e.pageX;
+    this.splitOffset_ = this.bar_./*OK*/getBoundingClientRect().left;
+  }
+  /**
+   * Handle drag move
+   * @param {Event} e
+   */
+  dragMove(e) {
+    e.preventDefault();
+    e.stopPropagation(); // avoid clashing with clickImage
+    this.pointerMoveX_(e.pageX);
+  }
+  /**
+   * Remove listeners on drag end
+   * e is optional since dragEnd will also be used for unregister cleanup
+   * @param {Event|null} e
+   */
+  dragEnd(e) {
+    if (e) {
+      e.stopPropagation();
+    }
+
+    this.win.removeEventListener('mousemove', this.dragMove);
+    this.win.removeEventListener('mouseup', this.dragEnd);
+
+    this.moveOffset_ = 0;
+    this.splitOffset_ = 0;
+  }
+  /**
+   * Add listeners on drag start
+   * @param {Event} e
+   */
+  touchStart(e) {
+    this.container_.addEventListener('touchmove', this.touchMove);
+    this.container_.addEventListener('touchend', this.touchEnd);
+
+    this.moveOffset_ = e.touches[0].pageX;
+    this.splitOffset_ = this.bar_./*OK*/getBoundingClientRect().left;
```

Comment:
Does it make sense to cache this value instead of calculate it every time? It won't actually change between the next and last interaction without informing this component anyway.